pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -535,6 +565,274 @@ func containerSucceeded(c *v1.Container, podStatus *kubecontainer.PodStatus) boo
+			return false, false
+		}
+		if resizePolicy[rName] == v1.RestartRequired {
+			return true, true
+		}
+		return true, false
+	}
+	markContainerForUpdate := func(rName ResizeResourceKind, specValue, statusValue int64) {
+		cUpdateInfo := containerToUpdateInfo{
+			apiContainer:        &pod.Spec.Containers[containerIdx],
+			apiContainerStatus:  &apiContainerStatus,
+			kubeContainerStatus: containerStatus,
+		}
+		// Order the container updates such that resource decreases are applied before increases
+		switch {
+		case specValue > statusValue: // append
+			changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], cUpdateInfo)
+		case specValue < statusValue: // prepend
+			changes.ContainersToUpdate[rName] = append(changes.ContainersToUpdate[rName], containerToUpdateInfo{})
+			copy(changes.ContainersToUpdate[rName][1:], changes.ContainersToUpdate[rName])
+			changes.ContainersToUpdate[rName][0] = cUpdateInfo
+		}
+	}
+	desiredLimits := container.Resources.Limits
+	desiredRequests := container.Resources.Requests
+	currentLimits := apiContainerStatus.Resources.Limits
+	currentRequests := apiContainerStatus.Resources.Requests
+	// Runtime container status resources, if set, takes precedence.
+	if containerStatus.Resources.Limits != nil {
+		currentLimits = containerStatus.Resources.Limits
+	}
+	if containerStatus.Resources.Requests != nil {
+		currentRequests = containerStatus.Resources.Requests
+	}
+	resizeCPULim, restartCPULim := determineContainerResize(v1.ResourceCPU, desiredLimits.Cpu().MilliValue(), currentLimits.Cpu().MilliValue())
+	resizeCPUReq, restartCPUReq := determineContainerResize(v1.ResourceCPU, desiredRequests.Cpu().MilliValue(), currentRequests.Cpu().MilliValue())
+	resizeMemLim, restartMemLim := determineContainerResize(v1.ResourceMemory, desiredLimits.Memory().Value(), currentLimits.Memory().Value())
+	if restartCPULim || restartCPUReq || restartMemLim {
+		// resize policy requires this container to restart
+		changes.ContainersToKill[containerStatus.ID] = containerToKillInfo{
+			name:      containerStatus.Name,
+			container: &pod.Spec.Containers[containerIdx],
+			message:   fmt.Sprintf("Container %s resize requires restart", container.Name),
+		}
+		changes.ContainersToStart = append(changes.ContainersToStart, containerIdx)
+		changes.UpdatePodResources = true
+		return false
+	} else {
+		if resizeCPULim {
+			markContainerForUpdate(ResizeCPULimit, desiredLimits.Cpu().MilliValue(), currentLimits.Cpu().MilliValue())
+		}
+		if resizeCPUReq {
+			markContainerForUpdate(ResizeCPURequest, desiredRequests.Cpu().MilliValue(), currentRequests.Cpu().MilliValue())
+		}
+		if resizeMemLim {
+			markContainerForUpdate(ResizeMemoryLimit, desiredLimits.Memory().Value(), currentLimits.Memory().Value())
+		}
+	}
+	return true
+}
+
+func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *kubecontainer.PodStatus, podContainerChanges podActions, result kubecontainer.PodSyncResult) {
```

Comment:
the result should be pointer

Reply:
Why do we need to return anything? What action can we take with any returned value besides logging?

Reply:
When an error occurs, I think the relevant err should be returned, which can be set on the pod resize status, such as Deferred or Error, so that we can perceive that an error has occurred in resize, and can send an event related to the error, so that we can feel the exception in time events, instead of looking at the logs.
When the subsequent syncPod retries, the resize may be successful, and the wrong resize status is cancelled at this time.